### ISSUE TITLE
L11 BRDF crash

### ISSUE THREAD

**Issue by oceancolorcoder** [nasa/HyperCP#213]
```
Applying Lee et al. 2011 BRDF correction to Rrs and nLw
Traceback (most recent call last):
  File "/accounts/daurin/Applications/anaconda3/envs/hypercp/lib/python3.11/site-packages/pandas/core/indexes/base.py", line 3802, in get_loc
    return self._engine.get_loc(casted_key)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "pandas/_libs/index.pyx", line 138, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 165, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1577, in pandas._libs.hashtable.Float64HashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1587, in pandas._libs.hashtable.Float64HashTable.get_item
KeyError: 665.0

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/accounts/daurin/Applications/anaconda3/envs/hypercp/lib/python3.11/site-packages/xarray/core/indexes.py", line 486, in sel
    indexer = self.index.get_loc(label_value)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/accounts/daurin/Applications/anaconda3/envs/hypercp/lib/python3.11/site-packages/pandas/core/indexes/base.py", line 3804, in get_loc
    raise KeyError(key) from err
KeyError: 665.0

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/accounts/daurin/GitRepos/HyperCP/Main.py", line 546, in multi2Clicked
    self.processMulti(2)
  File "/accounts/daurin/GitRepos/HyperCP/Main.py", line 540, in processMulti
    Controller.processFilesMultiLevel(self.outputDirectory,fileNames, calibrationMap, flag_Trios)
  File "/accounts/daurin/GitRepos/HyperCP/Source/Controller.py", line 815, in processFilesMultiLevel
    Controller.processSingleLevel(pathOut, fp, calibrationMap, 'L2', flag_Trios)
  File "/accounts/daurin/GitRepos/HyperCP/Source/Controller.py", line 736, in processSingleLevel
    root = Controller.processL2(root,outFilePath)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/accounts/daurin/GitRepos/HyperCP/Source/Controller.py", line 449, in processL2
    node = ProcessL2.processL2(root,station)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/accounts/daurin/GitRepos/HyperCP/Source/ProcessL2.py", line 2352, in processL2
    ProcessL2BRDF.procBRDF(node, BRDF_option='L11')
  File "/accounts/daurin/GitRepos/HyperCP/Source/ProcessL2BRDF.py", line 132, in procBRDF
    OC_BRDF = oc_brdf.brdf_prototype(xr_ds, brdf_model=BRDF_option)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/accounts/daurin/GitRepos/HyperCP/Source/ocbrdf/ocbrdf_main.py", line 73, in brdf_prototype
    ds = BRDF_model.backward(ds, iter_brdf)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/accounts/daurin/GitRepos/HyperCP/Source/ocbrdf/brdf_model_L11.py", line 126, in backward
    Rrs.loc[dict(bands=665)] = Rrs665
    ~~~~~~~^^^^^^^^^^^^^^^^^
  File "/accounts/daurin/Applications/anaconda3/envs/hypercp/lib/python3.11/site-packages/xarray/core/dataarray.py", line 222, in __setitem__
    dim_indexers = map_index_queries(self.data_array, key).dim_indexers
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/accounts/daurin/Applications/anaconda3/envs/hypercp/lib/python3.11/site-packages/xarray/core/indexing.py", line 190, in map_index_queries
    results.append(index.sel(labels, **options))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/accounts/daurin/Applications/anaconda3/envs/hypercp/lib/python3.11/site-packages/xarray/core/indexes.py", line 488, in sel
    raise KeyError(
KeyError: "not all values found in index 'bands'. Try setting the `method` keyword argument (example: method='nearest')."


```

**Comment by oceancolorcoder** [comment]
Offending raw files and configuration on the Box under Data

**Comment by juanchossn** [comment]
Hi Dirk,

I am testing this. However in my case it crashes before:

Filtering reflectance spectra for negative values.
100.0% of Rrs_HYPER spectra flagged
83.3% of nLw_HYPER spectra flagged
Removing records...
Remove REFLECTANCE Data
   Length of dataset prior to removal 6 long
Data group is empty. Continuing.
Data group is empty. Continuing.
Data group is empty. Continuing.
Data group is empty. Continuing.
Data group is empty. Continuing.
   Length of dataset after removal 0 long: 100% removed
Too few spectra remaining. Abort.

**Comment by juanchossn** [comment]
> Hi Dirk,
> 
> I am testing this. However in my case it crashes before:
> 
> Filtering reflectance spectra for negative values. 100.0% of Rrs_HYPER spectra flagged 83.3% of nLw_HYPER spectra flagged Removing records... Remove REFLECTANCE Data Length of dataset prior to removal 6 long Data group is empty. Continuing. Data group is empty. Continuing. Data group is empty. Continuing. Data group is empty. Continuing. Data group is empty. Continuing. Length of dataset after removal 0 long: 100% removed Too few spectra remaining. Abort.

--> Before could not reproduce likely because of issues in my conda environment in EUMETSAT infrastructure

**Comment by oceancolorcoder** [comment]
Resolved with 4f851000f88a04d47de3d7fa4f2279a2414f437a. Closing.
